### ISSUE TITLE
🐛 Bug: Migration mode should not clobber `.gitignore`

### ISSUE THREAD

**Issue by danvk** [JoshuaKGoldberg/create-typescript-app#1354]
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

For an existing project, the `.gitignore` file has probably accumulated lots of stuff that should be ignored.

I expected the migration script to add to `.gitignore` or adjust it, not to completely clobber it.

### Actual

Running `npx create-typescript-app` completely clobbers the `.gitignore` file, replacing it with a two-liner.

### Additional Info

I wound up with lots of previously-untracked files added to my repo after running the migrate script over at https://github.com/danvk/literate-ts/pull/246

For example: I was ignoring the `coverage` directory, which contained HTML reports of code coverage that I would periodically generate. After migration, git wanted to track this.

**Comment by JoshuaKGoldberg** [comment]
Agreed, makes sense. I think there should be a similar approach to how the migration handles existing dependencies: known no-longer-necessary ones should be removed, and other existing ones retained.
